File: 2015/09/1.rb
distances = Hash.new { |hash, key| hash[key] = {} }
answer = Float::INFINITY

File.open('input.txt').each do |line|
  start, finish, distance = line.match(/(\w+) to (\w+) = (\d+)/).captures
  distances[start][finish] = distance.to_i
  distances[finish][start] = distance.to_i
end

distances.keys.permutation.each do |permutation|
  distance = 0
  valid = true

  permutation.zip(permutation.rotate)[..-2].each do |start, finish|
    if distances[start].has_key?(finish)
      distance += distances[start][finish]
    else
      valid = false
      break
    end
  end

  answer = distance if distance < answer && valid
end

puts answer

File: 2022/04/1.py
from re import findall

answer = 0

with open("input.txt") as f:
    for line in f:
        a1, a2, b1, b2 = [int(m) for m in findall("(\d+)", line)]
        s1 = set(range(a1, a2+1))
        s2 = set(range(b1, b2+1))
        if s1.issubset(s2) or s2.issubset(s1):
            answer += 1

print(answer)
